daemon/exec.go

```diff
@@ -267,10 +267,13 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, stdin
 	// the exec context should be ready, or error happened.
 	// close the chan to notify readiness
 	close(ec.Started)
-	if err != nil {
+	if exitCode, err := translateContainerdStartErr(err); err != nil {
 		c.ExecCommands.Unlock()
+		if exitCode != -1 {
+			ec.SetExitCode(exitCode)
+		}
 		ec.Unlock()
```

Comment:
This moves the `Unlock()` to happen _after_ the exit-code was updated.

Possibly the `c.ExecCommands.Unlock()` should be moved as well (not sure if it matters when, and in which order they're unlocked)